types/react-dom/test-utils/index.d.ts

```diff
@@ -298,8 +298,9 @@ export function createRenderer(): ShallowRenderer;
 declare const UNDEFINED_VOID_ONLY: unique symbol;
 // tslint:disable-next-line: void-return
 type VoidOrUndefinedOnly = void | { [UNDEFINED_VOID_ONLY]: never };
+// While act does always return Thenable, if a void function is passed, the behavior is synchronous
```

Comment:
This is a bit inaccurate. The behavior of act doesn't change depending on void function or not (at runtime we can't even distinguish void and `undefined`.

The idea here is to make a concession to practicallity. If we return a Promise from void functions we would trigger dangling Promise lint rules. You can't chain `act` calls anyway and should just `await` them. And since you can `await` already unwrapped values anyway (i.e. `await act(() => {})` is perfectly fine at runtime) we just pretend the return value is not a Promise but `void`.
```suggestion
// While act does always return Thenable, if a void function is passed, we pretend the return value is also void to not trigger dangling Promise lint rules.
```